R/render-scorecard.R

```diff
@@ -206,21 +206,25 @@ map_answer <- function(scores, criteria, answer_breaks = c(0, 1)) {
 }
 
 
-#' Look for mitigation file and return contents if is found
+#' Look for comment file and return contents if is found
+#'
+#' The presence of a comment section indicates that we are aware the score is
+#' low, but we are proceeding with adding the package to MPN or otherwise
+#' approving it. This is not the same as a deviation from expected results.
 #'
 #' @inheritParams render_scorecard
 #'
 #' @keywords internal
-check_for_mitigation <- function(results_dir){
-  # mitigation (if any)
-  # infer mitigation path from `results_dir`
-  mitigation_path <- get_result_path(results_dir, "mitigation.txt")
-  if(fs::file_exists(mitigation_path)){
-    mitigation_block <- readLines(mitigation_path)
+check_for_comments <- function(results_dir){
+  # comments (if any)
+  # infer comments path from `results_dir`
+  comments_path <- get_result_path(results_dir, "comments.txt")
```

Comment:
Can we keep "mitigation.txt" around as a compatibility fallback?

Two reasons:

 * it avoids breaking scripts like `score-r-pkg`
 * even though there are unlikely to be previous users at this point, it's a bit surprising to have this file work previous and then have it silently ignored (or at least I'd guess that would be the behavior)

Reply:
Ill defer to @seth127 on this one. I think the section in the report is the main thing that had to change. We don't upload the text file to dropbox, so I dont think there's any way an auditor would see that file name.

I think it could be a little confusing to refer to it as 'comments' everywhere except the file extension and in `get_result_path()`, but if it makes a big difference on the `score-r-pkg` I can see motivation to revert that change.

Reply:
To expand a bit: I would prefer to add the fallback and think it's easy enough, but it's a judgement call and I'm all right with saying "nah, we don't want these sorts of compatibility kludges at this early stage".  In that case, it's still good to have this discussion in the review, and we can make sure to highlight this gotcha in the NEWS.

Reply:
That makes sense to me. (Sorry, I missed this point when I just approved a few minutes ago). Some questions I'd like input on:

* Do you think it's worth any kind of "deprecation warning" at this point whenever `-mitigation.txt` is used? 
* If only one or the other (i.e `-comments.txt` _or_ `-mitigation.txt`) are found, it seems obvious that we include the one that's found.
* If _both_ `-comments.txt` and `-mitigation.txt` are found, I can see a few options for the logic here. I'll list them in my order of preference, but very open to other thoughts.
  * Pull in `-comments.txt`, ignoring `-mitigation.txt` and warning (or messaging) the user to this effect.
  * Pull in `-comments.txt`, silently ignoring `-mitigation.txt`.
  * Concatenate `-comments.txt` and `-mitigation.txt` together and include both _(this feels like a bad idea to me, but I guess worth considering)_

Reply:
> Do you think it's worth any kind of "deprecation warning" at this point whenever `-mitigation.txt` is used?

Yeah, I think a warning saying to move to `-comments.txt` makes sense, mostly because then it lets us clean it up at some point without feeling like we need to first start giving a warning for a release *at that point*.

> If _both_ `-comments.txt` and `-mitigation.txt` are found, I can see a few options for the logic here. I'll list them in my order of preference, but very open to other thoughts.
>
> Pull in `-comments.txt`, ignoring `-mitigation.txt` and warning (or messaging) the user to this effect.

I agree with your preference for this.

---

And just to think through the migration on the score-r-pkg end a bit more:

 * I update it soon-ish to use comments.txt if mpn.scorecard is new enough

 * we tell people to update their script version

 * *but* if people don't update (and are scoring a package with mitigation/comments), it will still work but with a warning

Reply:
That makes sense to me. @barrettk do have clarity on how to go ahead with this change? 

You can consider using `lifecycle::deprecate_warn()`. Example from `bbr` in https://github.com/metrumresearchgroup/bbr/commit/5a6bc119b5fcc94086631b75ac767135df33b3c4 if you'd like a reference.